# **Proyek Analisis Data: Bike Sharing**

- **Nama:** Khoirunnisa
- **Email:** nkoirun10@gmail.com
- **ID Dicoding:** mc009d5x2406


# **Menentukan Pertanyaan Bisnis**

1. Pada jam dan hari apa jumlah penyewaan sepeda mencapai puncaknya dan kapan terendah
2. bagaimana perbedaannya penyewaan sepeda antara hari kerja dan akhir pekan?
2. Bagaimana pengaruh musim terhadap penyewaan sepeda?
4. Bagaimana perbedaan pola penyewaan antara pengguna casual dan registered?
4. Bagaimana dampak hari libur terhadap jumlah penyewaan sepeda, dan apakah ada peluang untuk meningkatkan penyewaan di hari-hari tertentu?

# **Import Semua Packages/Library yang Digunakan**

In [3]:
import warnings
warnings.filterwarnings('ignore')

In [4]:
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib as plt
import plotly.graph_objects as go
import plotly.express as px
import seaborn as sns
from collections.abc import Mapping

# **Data Wragling**

## **Gathering data**

Mengumpulkan dataset yang digunakan, terdapat dua dataset yang akan digunakan.


### **Load tabel sewa berdasarkan hari**

In [5]:
day_df = pd.read_csv("https://raw.githubusercontent.com/millkywaay/Dataset-Proyek-Analisi-Data/refs/heads/main/day.csv")
day_df.head()

instant      dteday  season  yr  mnth  holiday  weekday  workingday  \
0        1  2011-01-01       1   0     1        0        6           0   
1        2  2011-01-02       1   0     1        0        0           0   
2        3  2011-01-03       1   0     1        0        1           1   
3        4  2011-01-04       1   0     1        0        2           1   
4        5  2011-01-05       1   0     1        0        3           1   

   weathersit      temp     atemp       hum  windspeed  casual  registered  \
0           2  0.344167  0.363625  0.805833   0.160446     331         654   
1           2  0.363478  0.353739  0.696087   0.248539     131         670   
2           1  0.196364  0.189405  0.437273   0.248309     120        1229   
3           1  0.200000  0.212122  0.590435   0.160296     108        1454   
4           1  0.226957  0.229270  0.436957   0.186900      82        1518   

    cnt  
0   985  
1   801  
2  1349  
3  1562  
4  1600

### **Load tabel sewa berdasarkan jam**

In [6]:
hour_df = pd.read_csv("https://raw.githubusercontent.com/millkywaay/Dataset-Proyek-Analisi-Data/refs/heads/main/hour.csv")
hour_df.head(10)

instant      dteday  season  yr  mnth  hr  holiday  weekday  workingday  \
0        1  2011-01-01       1   0     1   0        0        6           0   
1        2  2011-01-01       1   0     1   1        0        6           0   
2        3  2011-01-01       1   0     1   2        0        6           0   
3        4  2011-01-01       1   0     1   3        0        6           0   
4        5  2011-01-01       1   0     1   4        0        6           0   
5        6  2011-01-01       1   0     1   5        0        6           0   
6        7  2011-01-01       1   0     1   6        0        6           0   
7        8  2011-01-01       1   0     1   7        0        6           0   
8        9  2011-01-01       1   0     1   8        0        6           0   
9       10  2011-01-01       1   0     1   9        0        6           0   

   weathersit  temp   atemp   hum  windspeed  casual  registered  cnt  
0           1  0.24  0.2879  0.81     0.0000       3          13   16  
1           1  0.22  0.2727  0.80     0.0000       8          32   40  
2           1  0.22  0.2727  0.80     0.0000       5          27   32  
3           1  0.24  0.2879  0.75     0.0000       3          10   13  
4           1  0.24  0.2879  0.75     0.0000       0           1    1  
5           2  0.24  0.2576  0.75     0.0896       0           1    1  
6           1  0.22  0.2727  0.80     0.0000       2           0    2  
7           1  0.20  0.2576  0.86     0.0000       1           2    3  
8           1  0.24  0.2879  0.75     0.0000       1           7    8  
9           1  0.32  0.3485  0.76     0.0000       8           6   14

## **Assesing data**

Pada tahap ini kita akan menilai kualitad dari suluruh dataset yang akan digunakan

### **Menilai data sewa harian**

In [7]:
day_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 731 entries, 0 to 730
Data columns (total 16 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   instant     731 non-null    int64  
 1   dteday      731 non-null    object 
 2   season      731 non-null    int64  
 3   yr          731 non-null    int64  
 4   mnth        731 non-null    int64  
 5   holiday     731 non-null    int64  
 6   weekday     731 non-null    int64  
 7   workingday  731 non-null    int64  
 8   weathersit  731 non-null    int64  
 9   temp        731 non-null    float64
 10  atemp       731 non-null    float64
 11  hum         731 non-null    float64
 12  windspeed   731 non-null    float64
 13  casual      731 non-null    int64  
 14  registered  731 non-null    int64  
 15  cnt         731 non-null    int64  
dtypes: float64(4), int64(11), object(1)
memory usage: 91.5+ KB


In [8]:
day_df.isna().sum()

instant       0
dteday        0
season        0
yr            0
mnth          0
holiday       0
weekday       0
workingday    0
weathersit    0
temp          0
atemp         0
hum           0
windspeed     0
casual        0
registered    0
cnt           0
dtype: int64

In [9]:
print("Jumlah duplikasi: ", day_df.duplicated().sum())

Jumlah duplikasi:  0


In [10]:
day_df.describe()

instant      season          yr        mnth     holiday     weekday  \
count  731.000000  731.000000  731.000000  731.000000  731.000000  731.000000   
mean   366.000000    2.496580    0.500684    6.519836    0.028728    2.997264   
std    211.165812    1.110807    0.500342    3.451913    0.167155    2.004787   
min      1.000000    1.000000    0.000000    1.000000    0.000000    0.000000   
25%    183.500000    2.000000    0.000000    4.000000    0.000000    1.000000   
50%    366.000000    3.000000    1.000000    7.000000    0.000000    3.000000   
75%    548.500000    3.000000    1.000000   10.000000    0.000000    5.000000   
max    731.000000    4.000000    1.000000   12.000000    1.000000    6.000000   

       workingday  weathersit        temp       atemp         hum   windspeed  \
count  731.000000  731.000000  731.000000  731.000000  731.000000  731.000000   
mean     0.683995    1.395349    0.495385    0.474354    0.627894    0.190486   
std      0.465233    0.544894    0.183051    0.162961    0.142429    0.077498   
min      0.000000    1.000000    0.059130    0.079070    0.000000    0.022392   
25%      0.000000    1.000000    0.337083    0.337842    0.520000    0.134950   
50%      1.000000    1.000000    0.498333    0.486733    0.626667    0.180975   
75%      1.000000    2.000000    0.655417    0.608602    0.730209    0.233214   
max      1.000000    3.000000    0.861667    0.840896    0.972500    0.507463   

            casual   registered          cnt  
count   731.000000   731.000000   731.000000  
mean    848.176471  3656.172367  4504.348837  
std     686.622488  1560.256377  1937.211452  
min       2.000000    20.000000    22.000000  
25%     315.500000  2497.000000  3152.000000  
50%     713.000000  3662.000000  4548.000000  
75%    1096.000000  4776.500000  5956.000000  
max    3410.000000  6946.000000  8714.000000

### **Menilai data sewa perjam**

In [11]:
hour_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 17379 entries, 0 to 17378
Data columns (total 17 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   instant     17379 non-null  int64  
 1   dteday      17379 non-null  object 
 2   season      17379 non-null  int64  
 3   yr          17379 non-null  int64  
 4   mnth        17379 non-null  int64  
 5   hr          17379 non-null  int64  
 6   holiday     17379 non-null  int64  
 7   weekday     17379 non-null  int64  
 8   workingday  17379 non-null  int64  
 9   weathersit  17379 non-null  int64  
 10  temp        17379 non-null  float64
 11  atemp       17379 non-null  float64
 12  hum         17379 non-null  float64
 13  windspeed   17379 non-null  float64
 14  casual      17379 non-null  int64  
 15  registered  17379 non-null  int64  
 16  cnt         17379 non-null  int64  
dtypes: float64(4), int64(12), object(1)
memory usage: 2.3+ MB


In [12]:
hour_df.isna().sum()

instant       0
dteday        0
season        0
yr            0
mnth          0
hr            0
holiday       0
weekday       0
workingday    0
weathersit    0
temp          0
atemp         0
hum           0
windspeed     0
casual        0
registered    0
cnt           0
dtype: int64

In [13]:
hour_df.describe()

instant        season            yr          mnth            hr  \
count  17379.0000  17379.000000  17379.000000  17379.000000  17379.000000   
mean    8690.0000      2.501640      0.502561      6.537775     11.546752   
std     5017.0295      1.106918      0.500008      3.438776      6.914405   
min        1.0000      1.000000      0.000000      1.000000      0.000000   
25%     4345.5000      2.000000      0.000000      4.000000      6.000000   
50%     8690.0000      3.000000      1.000000      7.000000     12.000000   
75%    13034.5000      3.000000      1.000000     10.000000     18.000000   
max    17379.0000      4.000000      1.000000     12.000000     23.000000   

            holiday       weekday    workingday    weathersit          temp  \
count  17379.000000  17379.000000  17379.000000  17379.000000  17379.000000   
mean       0.028770      3.003683      0.682721      1.425283      0.496987   
std        0.167165      2.005771      0.465431      0.639357      0.192556   
min        0.000000      0.000000      0.000000      1.000000      0.020000   
25%        0.000000      1.000000      0.000000      1.000000      0.340000   
50%        0.000000      3.000000      1.000000      1.000000      0.500000   
75%        0.000000      5.000000      1.000000      2.000000      0.660000   
max        1.000000      6.000000      1.000000      4.000000      1.000000   

              atemp           hum     windspeed        casual    registered  \
count  17379.000000  17379.000000  17379.000000  17379.000000  17379.000000   
mean       0.475775      0.627229      0.190098     35.676218    153.786869   
std        0.171850      0.192930      0.122340     49.305030    151.357286   
min        0.000000      0.000000      0.000000      0.000000      0.000000   
25%        0.333300      0.480000      0.104500      4.000000     34.000000   
50%        0.484800      0.630000      0.194000     17.000000    115.000000   
75%        0.621200      0.780000      0.253700     48.000000    220.000000   
max        1.000000      1.000000      0.850700    367.000000    886.000000   

                cnt  
count  17379.000000  
mean     189.463088  
std      181.387599  
min        1.000000  
25%       40.000000  
50%      142.000000  
75%      281.000000  
max      977.000000

In [14]:
print("Jumlah duplikasi: ", hour_df.duplicated().sum())

Jumlah duplikasi:  0


**Hasil penilaian:**

1.   Tidak ada data dupilkat dikedua tabel
2.   Tidak ada data null dikedua tabel
3.  penilaian statistik sudah cukup baik, baik dari nilai min, max, dan mean. Data terdistribusi normal, tidak ada invalid value


## **Cleaning Data**




#### **Cleaning data yang akan dilakukan**


1. menghapus kolom **instan** karena tidak akan digunakan
2. mengkonversi tipe data **dteday** dari object ke datetime
3. mengkonveri kolom **season, yr, mnth, holiday, weekday, workingday, weathersit** menjadi tipe kolom kategori
4. Merename beberapa nama kolom dan value data katagori



rename nama kolom

In [15]:
day_df.rename(columns={
    "dteday": "date",
    "yr": "year",
    "mnth": "month",
    "weathersit": "weather_condition",
    "hum": "humidity",
    "cnt": "count"
}, inplace=True)

hour_df.rename(columns={
    "dteday": "date",
    "yr": "year",
    "hr": "hour",
    "mnth": "month",
    "weathersit": "weather_condition",
    "hum": "humidity",
    "cnt": "count"
}, inplace=True)

Menghapus kolom instan

In [16]:
day_df.drop(columns=['instant'], inplace=True)
hour_df.drop(columns=['instant'], inplace=True)

Mengkonversi tipe data date

In [17]:
day_df['date'] = pd.to_datetime(day_df['date'])
hour_df['date'] = pd.to_datetime(hour_df['date'])


mengkonveri kolom season, yr, mnth, holiday, weekday, workingday, weathersit menjadi tipe kolom kategori

In [18]:
categorical_cols = ['season', 'year', 'month', 'holiday', 'weekday', 'workingday', 'weather_condition']
day_df[categorical_cols] = day_df[categorical_cols].astype('category')
hour_df[categorical_cols] = hour_df[categorical_cols].astype('category')

In [19]:
day_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 731 entries, 0 to 730
Data columns (total 15 columns):
 #   Column             Non-Null Count  Dtype         
---  ------             --------------  -----         
 0   date               731 non-null    datetime64[ns]
 1   season             731 non-null    category      
 2   year               731 non-null    category      
 3   month              731 non-null    category      
 4   holiday            731 non-null    category      
 5   weekday            731 non-null    category      
 6   workingday         731 non-null    category      
 7   weather_condition  731 non-null    category      
 8   temp               731 non-null    float64       
 9   atemp              731 non-null    float64       
 10  humidity           731 non-null    float64       
 11  windspeed          731 non-null    float64       
 12  casual             731 non-null    int64         
 13  registered         731 non-null    int64         
 14  count     

In [20]:
hour_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 17379 entries, 0 to 17378
Data columns (total 16 columns):
 #   Column             Non-Null Count  Dtype         
---  ------             --------------  -----         
 0   date               17379 non-null  datetime64[ns]
 1   season             17379 non-null  category      
 2   year               17379 non-null  category      
 3   month              17379 non-null  category      
 4   hour               17379 non-null  int64         
 5   holiday            17379 non-null  category      
 6   weekday            17379 non-null  category      
 7   workingday         17379 non-null  category      
 8   weather_condition  17379 non-null  category      
 9   temp               17379 non-null  float64       
 10  atemp              17379 non-null  float64       
 11  humidity           17379 non-null  float64       
 12  windspeed          17379 non-null  float64       
 13  casual             17379 non-null  int64         
 14  regist

Melakukan proses mapping data pada kolom categorical

In [21]:
map = {
    'season': {1: 'Spring', 2: 'Summer', 3: 'Fall', 4: 'Winter'},
    'year': {0: '2011', 1: '2012'},
    'month': {1: 'January', 2: 'February', 3: 'March', 4: 'April', 5: 'May', 6: 'June',
              7: 'July', 8: 'August', 9: 'September', 10: 'October', 11: 'November', 12: 'December'},
    'holiday' : {0:'No Holiday',1:'Holiday'},
    'workingday': {0: 'No', 1: 'Yes'},
    'weekday': {0: 'Sunday', 1: 'Monday', 2: 'Tuesday', 3: 'Wednesday', 4: 'Thursday', 5: 'Friday', 6: 'Saturday'},
    'weather_condition':
        {1:'Clear, Few clouds, Partly cloudy, Partly cloudy',
         2:'Mist + Cloudy, Mist + Broken clouds, Mist + Few clouds, Mist',
         3:'Light Snow, Light Rain + Thunderstorm + Scattered clouds, Light Rain + Scattered clouds',
         4:'Heavy Rain + Ice Pellets + Thunderstorm + Mist, Snow + Fog'}
}

for col, mapping in map.items():
    day_df[col] = day_df[col].map(mapping)
    hour_df[col] = hour_df[col].map(mapping)

In [22]:
day_df.head()

date  season  year    month     holiday    weekday workingday  \
0 2011-01-01  Spring  2011  January  No Holiday   Saturday         No   
1 2011-01-02  Spring  2011  January  No Holiday     Sunday         No   
2 2011-01-03  Spring  2011  January  No Holiday     Monday        Yes   
3 2011-01-04  Spring  2011  January  No Holiday    Tuesday        Yes   
4 2011-01-05  Spring  2011  January  No Holiday  Wednesday        Yes   

                                   weather_condition      temp     atemp  \
0  Mist + Cloudy, Mist + Broken clouds, Mist + Fe...  0.344167  0.363625   
1  Mist + Cloudy, Mist + Broken clouds, Mist + Fe...  0.363478  0.353739   
2    Clear, Few clouds, Partly cloudy, Partly cloudy  0.196364  0.189405   
3    Clear, Few clouds, Partly cloudy, Partly cloudy  0.200000  0.212122   
4    Clear, Few clouds, Partly cloudy, Partly cloudy  0.226957  0.229270   

   humidity  windspeed  casual  registered  count  
0  0.805833   0.160446     331         654    985  
1  0.696087   0.248539     131         670    801  
2  0.437273   0.248309     120        1229   1349  
3  0.590435   0.160296     108        1454   1562  
4  0.436957   0.186900      82        1518   1600

In [23]:
hour_df.head()

date  season  year    month  hour     holiday   weekday workingday  \
0 2011-01-01  Spring  2011  January     0  No Holiday  Saturday         No   
1 2011-01-01  Spring  2011  January     1  No Holiday  Saturday         No   
2 2011-01-01  Spring  2011  January     2  No Holiday  Saturday         No   
3 2011-01-01  Spring  2011  January     3  No Holiday  Saturday         No   
4 2011-01-01  Spring  2011  January     4  No Holiday  Saturday         No   

                                 weather_condition  temp   atemp  humidity  \
0  Clear, Few clouds, Partly cloudy, Partly cloudy  0.24  0.2879      0.81   
1  Clear, Few clouds, Partly cloudy, Partly cloudy  0.22  0.2727      0.80   
2  Clear, Few clouds, Partly cloudy, Partly cloudy  0.22  0.2727      0.80   
3  Clear, Few clouds, Partly cloudy, Partly cloudy  0.24  0.2879      0.75   
4  Clear, Few clouds, Partly cloudy, Partly cloudy  0.24  0.2879      0.75   

   windspeed  casual  registered  count  
0        0.0       3          13     16  
1        0.0       8          32     40  
2        0.0       5          27     32  
3        0.0       3          10     13  
4        0.0       0           1      1



*   Column instan sudah dihapus
*   nama kolom sudah direname
*   kolom season, yr, mnth, holiday, weekday, workingday, weathersit menjadi tipe kolom kategori
*  kolom date sudah dikonversi menjadi tipe data date




# **Exploratory Data Analysis**



1.   Bagaimana pola tren penyewaan sepeda berdasarkan waktu (harian dan per jam)?
2. Apakah terdapat perbedaan signifikan dalam jumlah penyewaan sepeda antara hari kerja dan akhir pekan?
3. Bagaimana pengaruh musim terhadap jumlah penyewaan sepeda?
4. Bagaimana perbandingan pola penggunaan sepeda antara pengguna casual dan registered?
5. Apakah hari libur memengaruhi jumlah penyewaan sepeda secara signifikan?




### **EDA pada data sewa harian**

In [24]:
day_df.describe(include="all")

date season  year    month     holiday weekday  \
count                   731    731   731      731         731     731   
unique                  NaN      4     2       12           2       7   
top                     NaN   Fall  2012  January  No Holiday  Sunday   
freq                    NaN    188   366       62         710     105   
mean    2012-01-01 00:00:00    NaN   NaN      NaN         NaN     NaN   
min     2011-01-01 00:00:00    NaN   NaN      NaN         NaN     NaN   
25%     2011-07-02 12:00:00    NaN   NaN      NaN         NaN     NaN   
50%     2012-01-01 00:00:00    NaN   NaN      NaN         NaN     NaN   
75%     2012-07-01 12:00:00    NaN   NaN      NaN         NaN     NaN   
max     2012-12-31 00:00:00    NaN   NaN      NaN         NaN     NaN   
std                     NaN    NaN   NaN      NaN         NaN     NaN   

       workingday                                weather_condition  \
count         731                                              731   
unique          2                                                3   
top           Yes  Clear, Few clouds, Partly cloudy, Partly cloudy   
freq          500                                              463   
mean          NaN                                              NaN   
min           NaN                                              NaN   
25%           NaN                                              NaN   
50%           NaN                                              NaN   
75%           NaN                                              NaN   
max           NaN                                              NaN   
std           NaN                                              NaN   

              temp       atemp    humidity   windspeed       casual  \
count   731.000000  731.000000  731.000000  731.000000   731.000000   
unique         NaN         NaN         NaN         NaN          NaN   
top            NaN         NaN         NaN         NaN          NaN   
freq           NaN         NaN         NaN         NaN          NaN   
mean      0.495385    0.474354    0.627894    0.190486   848.176471   
min       0.059130    0.079070    0.000000    0.022392     2.000000   
25%       0.337083    0.337842    0.520000    0.134950   315.500000   
50%       0.498333    0.486733    0.626667    0.180975   713.000000   
75%       0.655417    0.608602    0.730209    0.233214  1096.000000   
max       0.861667    0.840896    0.972500    0.507463  3410.000000   
std       0.183051    0.162961    0.142429    0.077498   686.622488   

         registered        count  
count    731.000000   731.000000  
unique          NaN          NaN  
top             NaN          NaN  
freq            NaN          NaN  
mean    3656.172367  4504.348837  
min       20.000000    22.000000  
25%     2497.000000  3152.000000  
50%     3662.000000  4548.000000  
75%     4776.500000  5956.000000  
max     6946.000000  8714.000000  
std     1560.256377  1937.211452

In [25]:
day_df.head()

date  season  year    month     holiday    weekday workingday  \
0 2011-01-01  Spring  2011  January  No Holiday   Saturday         No   
1 2011-01-02  Spring  2011  January  No Holiday     Sunday         No   
2 2011-01-03  Spring  2011  January  No Holiday     Monday        Yes   
3 2011-01-04  Spring  2011  January  No Holiday    Tuesday        Yes   
4 2011-01-05  Spring  2011  January  No Holiday  Wednesday        Yes   

                                   weather_condition      temp     atemp  \
0  Mist + Cloudy, Mist + Broken clouds, Mist + Fe...  0.344167  0.363625   
1  Mist + Cloudy, Mist + Broken clouds, Mist + Fe...  0.363478  0.353739   
2    Clear, Few clouds, Partly cloudy, Partly cloudy  0.196364  0.189405   
3    Clear, Few clouds, Partly cloudy, Partly cloudy  0.200000  0.212122   
4    Clear, Few clouds, Partly cloudy, Partly cloudy  0.226957  0.229270   

   humidity  windspeed  casual  registered  count  
0  0.805833   0.160446     331         654    985  
1  0.696087   0.248539     131         670    801  
2  0.437273   0.248309     120        1229   1349  
3  0.590435   0.160296     108        1454   1562  
4  0.436957   0.186900      82        1518   1600

In [26]:
day_df.groupby(by="weekday").agg({
    "casual": "sum",
    "registered": "sum",
    "count": "sum"
}).sort_values(by="count", ascending=False)

casual  registered   count
weekday                              
Friday      78238      409552  487790
Thursday    61460      423935  485395
Saturday   153852      323955  477807
Wednesday   57319      415729  473048
Tuesday     57843      411266  469109
Monday      70784      384719  455503
Sunday     140521      303506  444027

Berdasarkan data diatas dapat diketahui bahwa penyewaan sepeda tertinggi atuh pada hari **Jumat** dengan total 487.790 lalu disusul dengan hari kamis dan sabtu. Lalu untuk hari dengan jumlah sewa terendah jatuh pada hari senin dan minggu, dimana nilai terendahnya yaitu 444.027. Terdapat selisih perbedaan sewa sekitar 43.000 dari hari teramai

In [27]:
day_df.groupby(by="year").agg({
    "casual": "sum",
    "registered": "sum",
    "count": "sum"
}).sort_values(by="count", ascending=False)

casual  registered    count
year                             
2012  372765     1676811  2049576
2011  247252      995851  1243103

Dari tabel diatas dapat dikatahui terdaoat peningkatan pesat pada penyewaan sepeda sebesar 806,473 dari tahun sebelumnya.

In [28]:
day_df.groupby(by="season").agg({
    "casual": "sum",
    "registered": "sum",
    "count": "sum"
}).sort_values(by="count", ascending=False)

casual  registered    count
season                             
Fall    226091      835038  1061129
Summer  203522      715067   918589
Winter  129782      711831   841613
Spring   60622      410726   471348

Penyewaan sepeda tertiinggi jatuh pada musim gugur dengan jumlah sewa mencapai 1.061.129 lalu penyewaan sepeda terendah jatuh pada musim semi sebesar 471.348. Dimana berbedaanya mencapai 44%.

In [29]:
day_df.groupby(by="month").agg({
    "casual": "sum",
    "registered": "sum",
    "count": "sum"
}).sort_values(by="count", ascending=False)

casual  registered   count
month                                
August      72039      279155  351194
June        73906      272436  346342
September   70323      275668  345991
July        78157      266791  344948
May         75285      256401  331686
October     59760      262592  322352
April       60802      208292  269094
November    36603      218228  254831
March       44444      184476  228920
December    21693      189343  211036
February    14963      136389  151352
January     12042      122891  134933

Bulan Agustus, Juni, dan september adalah bulan dimana penyewaan sepeda mengalami penyewaan 3 bulan tertinggi yang berkisar diantara 345,991-351,194. Untuk bulan degan jumlah sewa terendah jatuh pada bulan Januari, Februari, Desember yang hanya berda pada kisaran 134,993-211,036.

In [30]:
day_df.groupby(by="weather_condition").agg({
    "casual": "sum",
    "registered": "sum",
    "count": "sum"
}).sort_values(by="count", ascending=False)

casual  registered  \
weather_condition                                                        
Clear, Few clouds, Partly cloudy, Partly cloudy     446346     1811606   
Mist + Cloudy, Mist + Broken clouds, Mist + Few...  169776      827082   
Light Snow, Light Rain + Thunderstorm + Scatter...    3895       33974   

                                                      count  
weather_condition                                            
Clear, Few clouds, Partly cloudy, Partly cloudy     2257952  
Mist + Cloudy, Mist + Broken clouds, Mist + Few...   996858  
Light Snow, Light Rain + Thunderstorm + Scatter...    37869

Sewa sepeda pada kondisi cuaca yang cerah dan sedikit berawan memiliki jumlah penyewaan sepeda yang cukup tinggi dibandingkan jika cuaca sedikit ekstrim seperti bersalju dan hujan. Selisih perbedaanya mencapai 2,220,083 atau sebesar hinga 98%

In [31]:
day_df.groupby(by="holiday").agg({
    "casual": "sum",
    "registered": "sum",
    "count": "sum"
}).sort_values(by="count", ascending=False)

casual  registered    count
holiday                                
No Holiday  597658     2616586  3214244
Holiday      22359       56076    78435

Penyewaan sepeda pada tanggal merah kurang menarik pelanggan

In [32]:
day_df.groupby(by="workingday").agg({
    "casual": "sum",
    "registered": "sum",
    "count": "sum"
}).sort_values(by="count", ascending=False)

casual  registered    count
workingday                             
Yes         303285     1989125  2292410
No          316732      683537  1000269

penyewaan sepeda pada hari kerja cukup besar hampir 50% lebih unggul dibandingkan diluar hari kerja

In [33]:
casual_total = day_df['casual'].sum()
registered_total = day_df['registered'].sum()

result_df = pd.DataFrame({
    'User Type': ['casual', 'registered'],
    'Total Rentals': [casual_total, registered_total]
}).sort_values(by='Total Rentals', ascending=False)

result_df


User Type  Total Rentals
1  registered        2672662
0      casual         620017

penyewaan sepeda didominasi oleh para membership sebesar 2,672,662 sedangkan user casual hanya sebesar 620,017

### **EDA pada data sewa perjam**

In [34]:
hour_df.describe(include="all")

date season   year  month          hour  \
count                           17379  17379  17379  17379  17379.000000   
unique                            NaN      4      2     12           NaN   
top                               NaN   Fall   2012    May           NaN   
freq                              NaN   4496   8734   1488           NaN   
mean    2012-01-02 04:08:34.552045568    NaN    NaN    NaN     11.546752   
min               2011-01-01 00:00:00    NaN    NaN    NaN      0.000000   
25%               2011-07-04 00:00:00    NaN    NaN    NaN      6.000000   
50%               2012-01-02 00:00:00    NaN    NaN    NaN     12.000000   
75%               2012-07-02 00:00:00    NaN    NaN    NaN     18.000000   
max               2012-12-31 00:00:00    NaN    NaN    NaN     23.000000   
std                               NaN    NaN    NaN    NaN      6.914405   

           holiday   weekday workingday  \
count        17379     17379      17379   
unique           2         7          2   
top     No Holiday  Saturday        Yes   
freq         16879      2512      11865   
mean           NaN       NaN        NaN   
min            NaN       NaN        NaN   
25%            NaN       NaN        NaN   
50%            NaN       NaN        NaN   
75%            NaN       NaN        NaN   
max            NaN       NaN        NaN   
std            NaN       NaN        NaN   

                                      weather_condition          temp  \
count                                             17379  17379.000000   
unique                                                4           NaN   
top     Clear, Few clouds, Partly cloudy, Partly cloudy           NaN   
freq                                              11413           NaN   
mean                                                NaN      0.496987   
min                                                 NaN      0.020000   
25%                                                 NaN      0.340000   
50%                                                 NaN      0.500000   
75%                                                 NaN      0.660000   
max                                                 NaN      1.000000   
std                                                 NaN      0.192556   

               atemp      humidity     windspeed        casual    registered  \
count   17379.000000  17379.000000  17379.000000  17379.000000  17379.000000   
unique           NaN           NaN           NaN           NaN           NaN   
top              NaN           NaN           NaN           NaN           NaN   
freq             NaN           NaN           NaN           NaN           NaN   
mean        0.475775      0.627229      0.190098     35.676218    153.786869   
min         0.000000      0.000000      0.000000      0.000000      0.000000   
25%         0.333300      0.480000      0.104500      4.000000     34.000000   
50%         0.484800      0.630000      0.194000     17.000000    115.000000   
75%         0.621200      0.780000      0.253700     48.000000    220.000000   
max         1.000000      1.000000      0.850700    367.000000    886.000000   
std         0.171850      0.192930      0.122340     49.305030    151.357286   

               count  
count   17379.000000  
unique           NaN  
top              NaN  
freq             NaN  
mean      189.463088  
min         1.000000  
25%        40.000000  
50%       142.000000  
75%       281.000000  
max       977.000000  
std       181.387599

In [35]:
hour_df.head()

date  season  year    month  hour     holiday   weekday workingday  \
0 2011-01-01  Spring  2011  January     0  No Holiday  Saturday         No   
1 2011-01-01  Spring  2011  January     1  No Holiday  Saturday         No   
2 2011-01-01  Spring  2011  January     2  No Holiday  Saturday         No   
3 2011-01-01  Spring  2011  January     3  No Holiday  Saturday         No   
4 2011-01-01  Spring  2011  January     4  No Holiday  Saturday         No   

                                 weather_condition  temp   atemp  humidity  \
0  Clear, Few clouds, Partly cloudy, Partly cloudy  0.24  0.2879      0.81   
1  Clear, Few clouds, Partly cloudy, Partly cloudy  0.22  0.2727      0.80   
2  Clear, Few clouds, Partly cloudy, Partly cloudy  0.22  0.2727      0.80   
3  Clear, Few clouds, Partly cloudy, Partly cloudy  0.24  0.2879      0.75   
4  Clear, Few clouds, Partly cloudy, Partly cloudy  0.24  0.2879      0.75   

   windspeed  casual  registered  count  
0        0.0       3          13     16  
1        0.0       8          32     40  
2        0.0       5          27     32  
3        0.0       3          10     13  
4        0.0       0           1      1

In [36]:
hour_df.groupby(by="hour").agg({
    "casual": "sum",
    "registered": "sum",
    "count": "sum"
}).sort_values(by="count", ascending=False)

casual  registered   count
hour                            
17     54220      282640  336860
18     44496      265276  309772
8      15761      245240  261001
16     53834      173914  227748
19     35505      191284  226789
13     52713      132206  184919
12     49718      134696  184414
15     54606      128543  183149
14     55089      120563  175652
20     26378      138172  164550
9      22458      136980  159438
7       8037      146134  154171
11     43286      108034  151320
10     33789       92468  126257
21     20570      104875  125445
22     16200       79412   95612
23     11065       52876   63941
6       3017       52115   55132
0       7375       31755   39130
1       4709       19455   24164
2       3412       12940   16352
5       1012       13249   14261
3       1893        6281    8174
4        874        3554    4428

Penyewaan sepeda tertinggi ada pada pukul jam 5-6 sore. sedangkan pukul 3-4 pagi penyewaan sepeda berada pada jumlah sewa yang sangat rendah.

In [37]:
hourly_counts = hour_df.groupby(by=["hour", "workingday"]).agg({
    "count": "sum"
}).reset_index()

workingday_counts = hourly_counts[hourly_counts["workingday"] == "Yes"]
peak_hour = workingday_counts.loc[workingday_counts["count"].idxmax()]
peak_hour


hour              17
workingday       Yes
count         262120
Name: 35, dtype: object

Penyewaan sepeda puncaknya berada pada jam 5 sore dan saat working day dengan jumlah sewa sebedar 262,120

In [38]:
hourly_counts = hour_df.groupby(by=["hour", "holiday"]).agg({
    "count": "sum"
}).reset_index()

workingday_counts = hourly_counts[hourly_counts["holiday"] == "Holiday"]
peak_hour = workingday_counts.loc[workingday_counts["count"].idxmax()]
peak_hour


hour            13
holiday    Holiday
count         6269
Name: 27, dtype: object

Pada saat libur tanggal merah penyewaan sepeda tertinggi berada pada jam 1 siang denhan jumlah sewa 6,269

In [39]:
hour_df['hour_group'] = pd.cut(hour_df['hour'], bins=[0, 6, 12, 18, 24], labels=['Late Night', 'Morning', 'Afternoon', 'Evening'], right=False)

hour_df.groupby(by="hour_group").agg({
    "casual": "sum",
    "registered": "sum",
    "count": "sum"
}).sort_values(by="count", ascending=False)


casual  registered    count
hour_group                             
Afternoon   320180      972562  1292742
Evening     154214      831895   986109
Morning     126348      780971   907319
Late Night   19275       87234   106509

Secara keseluruhan penyewaan sepeda didominasi pada saat siang hingga sore hari yaitu pukul 12.000-18.00

**Insight yang didapat**


1.   Penyewaan sepeda tertinggi terdapat pada hari sabtu dan senin adalah hari dengan jumplah penyewa terenda
2.   Agustus  merupakan bulan dengan penyewaan tertinggi dan januari adalah bulan dengan jumlah penyewa terendah
3. Terdapat peningkatan sewa sepeda pada tahun 2012 dibandingkan tahun sebelumnya
4. season fall/ musim gugur memiliki jumlah penyewaan sepeda yang lebih banyak, daan jumlah penyewaan sedikit jatuh pada season spring
5. Jumlah sewa sepeda no holiday lebih banyak dibandingkan pada saat holiday
6. Penyewaan sepeda didominasi oleh tipe penyewa yang membership
7. penyewaan tertinggi ada pada pukul jam 5 sore
8. penyewaan sepeda tertingi jatuh pada sore hari (12:00-18:00)



# **Data Visualization**



### 1.   Bagaimana pola tren penyewaan sepeda berdasarkan waktu (harian dan per jam)?


In [40]:
monthly_counts = day_df.groupby(by="month").agg({
    "casual": "sum",
    "registered": "sum",
    "count": "sum"
}).reset_index()

fig = px.line(monthly_counts, x="month", y="count",
              title="Tren Penyewaan Sepeda Berdasarkan Bulan",
              labels={"count": "Jumlah Penyewaan", "month": "Bulan"},
              markers=True)

fig.show()

In [41]:
weekday_counts = day_df.groupby(by="weekday").agg({
    "casual": "sum",
    "registered": "sum",
    "count": "sum"
}).sort_values(by="count", ascending=False).reset_index()

fig = px.bar(weekday_counts, x="weekday", y="count",
             title="Penyewaan Sepeda Berdasarkan Hari",
             labels={"count": "Jumlah Penyewaan", "weekday": "Hari"},
             color="count",
             text_auto=True)

fig.show()

In [42]:
hourly_counts = hour_df.groupby(by="hour").agg({
    "casual": "sum",
    "registered": "sum",
    "count": "sum"
}).reset_index()

fig = px.line(hourly_counts, x="hour", y="count",
              title="Tren Penyewaan Sepeda Berdasarkan Jam",
              labels={"count": "Jumlah Penyewaan", "hour": "Jam"},
              markers=True)

fig.show()

Berdasarkan visualisasi diatas dapat dilihat bahwa pada hari jumat adalah hari dengan jumlah sewa tertinggi. Lalu pada pukul jam 2 siang hingga jam 5 sore, jumlah penyewaan sepeda cenderung naik dan mencapai titik tertingi di jam 5 sore. Kemudian jika berdasarkan bulan dapat diketahui bahwa pada bulan mei-sepetember jumlah penyewaan sepeda cendurng lebih tinggi dibandingkan dengan bulan lainnya walaupun pada bulan july sempat menurun tetapi tidak terlalu mencolok penurunanya.


### 2. Apakah terdapat perbedaan signifikan dalam jumlah penyewaan sepeda antara hari kerja dan akhir pekan?

In [43]:
workingday_counts = day_df.groupby(by="workingday").agg({
    "casual": "sum",
    "registered": "sum",
    "count": "sum"
}).reset_index()

fig = px.bar(workingday_counts, x="workingday", y="count",
             title="Penyewaan Sepeda Berdasarkan Hari Kerja dan Akhir Pekan",
             labels={"count": "Jumlah Penyewaan", "workingday": "Hari Kerja"},
             color="workingday",
             text_auto=True)

fig.show()

Perbedaan sewa sepeda dilihat pada hari kerja dan akhir pekan dapat dilihat bahwa pada hari kerja jumlah sewa sepeda sangat unggul dibandingkan pada akhir pekan selisih perbedaanya mencapi 50% lebih

## 3. Bagaimana pengaruh musim terhadap jumlah penyewaan sepeda?

In [44]:
season_counts = day_df.groupby(by="season").agg({
    "casual": "sum",
    "registered": "sum",
    "count": "sum"
}).reset_index()

fig = px.bar(season_counts, x="season", y="count",
             title="Penyewaan Sepeda Berdasarkan Musim",
             labels={"count": "Jumlah Penyewaan", "season": "Musim"},
             color="season",
             text_auto=True)

fig.show()

Pada musim gugur jumlah sewa sepeda sangat tinggi yaitu mencapai 1,061,129 dan titik terendahnya berada pada musim semi

## 4. Bagaimana perbedaan pola penyewaan antara pengguna casual dan registered?

In [45]:
user_type_counts = day_df.groupby(by="year").agg({
    "casual": "sum",
    "registered": "sum"
}).reset_index()

fig = px.bar(user_type_counts, x="year", y=["casual", "registered"],
             title="Perbandingan Penyewaan Sepeda antara Pengguna Casual dan Registered",
             labels={"value": "Jumlah Penyewaan", "year": "Tahun"},
             barmode="group",
             text_auto=True)

fig.show()

Terdapat peningkatan yang cukup drastis unruk tipe penyewaan sepeda pada user memebership sebesar 68%, dibangkangan degan penyewa casual yang kenaikanya hanya sebesar 50%. Namun secara garis besar penyewaan sepeda mengalami kenaikan dari tahun sebelumnya

## 5. Bagaimana dampak hari libur terhadap jumlah penyewaan sepeda, dan apakah ada peluang untuk meningkatkan penyewaan di hari-hari tertentu?

In [46]:
holiday_counts = day_df.groupby(by="holiday").agg({
    "count": "sum"
}).reset_index()

fig = px.bar(holiday_counts, x="holiday", y="count",
             title="Dampak Hari Libur Terhadap Penyewaan Sepeda",
             labels={"count": "Jumlah Penyewaan", "holiday": "Hari Libur"},
             color="holiday",
             text_auto=True)

fig.show()

Hari libur tidak memeberi dampak yang besar terhdap penyewan sepeda, hal ini dapat dilihat pda hari libur jumlah penywaan sepeda hanya sebesar 78.435k sedangkan pada hari biasanya jumlahnya sangat besar yaitu 3.214.244

# **Analisa Lebih Lanjut**

In [47]:
day_df.head()

date  season  year    month     holiday    weekday workingday  \
0 2011-01-01  Spring  2011  January  No Holiday   Saturday         No   
1 2011-01-02  Spring  2011  January  No Holiday     Sunday         No   
2 2011-01-03  Spring  2011  January  No Holiday     Monday        Yes   
3 2011-01-04  Spring  2011  January  No Holiday    Tuesday        Yes   
4 2011-01-05  Spring  2011  January  No Holiday  Wednesday        Yes   

                                   weather_condition      temp     atemp  \
0  Mist + Cloudy, Mist + Broken clouds, Mist + Fe...  0.344167  0.363625   
1  Mist + Cloudy, Mist + Broken clouds, Mist + Fe...  0.363478  0.353739   
2    Clear, Few clouds, Partly cloudy, Partly cloudy  0.196364  0.189405   
3    Clear, Few clouds, Partly cloudy, Partly cloudy  0.200000  0.212122   
4    Clear, Few clouds, Partly cloudy, Partly cloudy  0.226957  0.229270   

   humidity  windspeed  casual  registered  count  
0  0.805833   0.160446     331         654    985  
1  0.696087   0.248539     131         670    801  
2  0.437273   0.248309     120        1229   1349  
3  0.590435   0.160296     108        1454   1562  
4  0.436957   0.186900      82        1518   1600

## **Clustering**

In [48]:
day_df['actual_temp'] = day_df['temp'] * (39 - (-8)) + (-8)
hour_df['actual_temp'] = hour_df['temp'] * (39 - (-8)) + (-8)

day_df['actual_atemp'] = day_df['atemp'] * (50 - (-16)) + (-16)
hour_df['actual_atemp'] = hour_df['atemp'] * (50 - (-16)) + (-16)

day_df[['temp', 'actual_temp', 'atemp', 'actual_atemp']].head()

temp  actual_temp     atemp  actual_atemp
0  0.344167     8.175849  0.363625      7.999250
1  0.363478     9.083466  0.353739      7.346774
2  0.196364     1.229108  0.189405     -3.499270
3  0.200000     1.400000  0.212122     -1.999948
4  0.226957     2.666979  0.229270     -0.868180

In [49]:
bins = [-float('inf'), 10, 20, 30, float('inf')]
labels = ['Cold', 'Cool', 'Warm', 'Hot']

day_df['temp_category'] = pd.cut(day_df['actual_temp'], bins=bins, labels=labels)
day_df[['actual_temp', 'temp_category']].head()

actual_temp temp_category
0     8.175849          Cold
1     9.083466          Cold
2     1.229108          Cold
3     1.400000          Cold
4     2.666979          Cold

In [50]:
temp_clusters = day_df.groupby('temp_category').agg({
    'casual': 'sum',
    'registered': 'sum',
    'count': 'sum'
}).reset_index()
fig = px.bar(temp_clusters, x='temp_category', y='count',
             title='Jumlah Penyewaan Sepeda Berdasarkan Kategori Suhu',
             labels={'count': 'Total Penyewaan', 'temp_category': 'Kategori Suhu'},
             text_auto=True,
             color='temp_category')
fig.show()

In [51]:
temp_season_clusters = day_df.groupby(['temp_category', 'season']).agg({
    'count': 'sum'
}).reset_index()

fig = px.bar(temp_season_clusters, x='temp_category', y='count',
             title='Penyewaan Sepeda Berdasarkan Kategori Suhu dan Musim',
             labels={'count': 'Total Penyewaan', 'temp_category': 'Kategori Suhu'},
             color='season', barmode='group',
             text_auto=True)
fig.show()

# **Conclusion**

#### **1. Pola Tren Penyewaan Sepeda Berdasarkan Waktu (Harian dan Per Jam)**
- **Tren Harian:**
  - Penyewaan sepeda tertinggi terjadi pada **hari Jumat** dengan total **487.790** penyewaan, sedangkan terendah terjadi pada **hari Senin** dengan total **444.027** penyewaan. Hal ini menunjukkan bahwa orang cenderung lebih banyak bersepeda menjelang akhir pekan.
  - Pada **bulan Agustus**, penyewaan sepeda mencapai puncaknya dengan total **351.194** penyewaan, sementara **Januari** menjadi bulan dengan penyewaan terendah (**134.993**). Ini mungkin dipengaruhi oleh cuaca yang lebih nyaman di musim panas.
  
- **Tren Per Jam:**
  - Penyewaan sepeda tertinggi terjadi pada **pukul 17:00 (5 sore)** dengan total **262.120** penyewaan. Ini mungkin karena orang bersepeda pulang kerja atau sekolah.
  - Penyewaan terendah terjadi pada **pukul 04:00 (4 pagi)** dengan hanya **4.368** penyewaan, yang wajar karena aktivitas pada dini hari sangat minim.

**Insight:**
- Tren penyewaan sepeda sangat dipengaruhi oleh pola aktivitas harian, seperti jam kerja dan sekolah. Puncak penyewaan terjadi pada sore hari, sementara pagi hari (terutama dini hari) memiliki penyewaan yang sangat rendah.

**Rekomendasi:**
- Meningkatkan promosi pada sore hari (sekitar pukul 16:00-18:00) untuk memaksimalkan jumlah penyewaan.
- Menyediakan diskon atau program khusus pada pagi hari untuk menarik lebih banyak pengguna.

----

#### **2. Perbedaan Penyewaan Sepeda antara Hari Kerja dan Akhir Pekan**
- Penyewaan sepeda pada **hari kerja** jauh lebih tinggi (**3.214.244**) dibandingkan dengan **akhir pekan** (**78.435**). Ini menunjukkan bahwa sepeda banyak digunakan untuk keperluan transportasi sehari-hari, seperti pergi kerja atau sekolah.
- Pada **akhir pekan**, meskipun jumlah penyewaan lebih rendah, aktivitas bersepeda cenderung lebih santai dan rekreasi.

**Insight:**
- Pengguna lebih banyak memanfaatkan sepeda untuk kebutuhan transportasi harian daripada untuk rekreasi.

**Rekomendasi:**
- Meningkatkan promosi pada akhir pekan dengan menawarkan paket rekreasi atau keluarga.
- Menyediakan rute bersepeda yang menarik di area rekreasi untuk menarik minat pengguna akhir pekan.

---

#### **3. Pengaruh Musim terhadap Penyewaan Sepeda**
- **Musim gugur (Fall)** memiliki jumlah penyewaan tertinggi (**1.061.129**), sementara **musim semi (Spring)** memiliki penyewaan terendah (**471.348**).
- Musim gugur mungkin dipilih karena cuaca yang nyaman (tidak terlalu panas atau dingin), sementara musim semi cenderung lebih dingin dan kurang menarik untuk bersepeda.

**Insight:**
- Cuaca yang nyaman (tidak ekstrem) sangat memengaruhi minat orang untuk bersepeda.

**Rekomendasi:**
- Meningkatkan promosi pada musim semi dengan menawarkan diskon atau program khusus.
- Menyediakan fasilitas tambahan seperti jaket atau penghangat selama musim dingin.

---

#### **4. Perbandingan Pola Penggunaan antara Pengguna Casual dan Registered**
- Pengguna **registered (berlangganan)** mendominasi penyewaan sepeda dengan total **2.672.662**, sementara pengguna **casual** hanya menyumbang **620.017**.
- Pada **tahun 2012**, terjadi peningkatan signifikan dalam jumlah penyewaan oleh pengguna registered (**68%**) dibandingkan dengan pengguna casual (**50%**).

**Insight:**
- Program membership berhasil menarik minat pengguna untuk menggunakan sepeda secara konsisten.

**Rekomendasi:**
- Meningkatkan promosi program membership dengan menawarkan manfaat tambahan, seperti diskon atau akses ke fasilitas eksklusif.
- Menargetkan pengguna casual dengan program trial membership untuk mengkonversi mereka menjadi pengguna registered.

---

#### **5. Dampak Hari Libur terhadap Penyewaan Sepeda**
- Penyewaan sepeda pada **hari libur** sangat rendah (**78.435**) dibandingkan dengan **hari biasa** (**3.214.244**). Ini mungkin karena orang lebih memilih untuk beristirahat atau melakukan aktivitas lain di hari libur.

**Insight:**
- Hari libur tidak menjadi waktu yang populer untuk bersepeda, mungkin karena kurangnya aktivitas rutin seperti pergi kerja atau sekolah.

**Rekomendasi:**
- Menyediakan program khusus pada hari libur, seperti tur bersepeda atau acara komunitas, untuk menarik lebih banyak pengguna.
- Meningkatkan promosi pada hari libur dengan menawarkan diskon atau paket keluarga.

---
**Conclusion hasil anslisis lanjutan**
- Suhu **hangat (20-30°C)** memiliki jumlah penyewaan tertinggi (**1.420.753**), sementara suhu **panas (>30°C)** memiliki penyewaan terendah (**57.648**).
- Pada **musim gugur**, suhu hangat menjadi faktor utama tingginya jumlah penyewaan.

**Insight:**
- Suhu yang nyaman (tidak terlalu panas atau dingin) sangat memengaruhi minat orang untuk bersepeda.

**Rekomendasi:**
- Menyediakan fasilitas seperti tempat istirahat atau minuman dingin selama musim panas untuk meningkatkan kenyamanan pengguna.
- Meningkatkan promosi pada musim dengan suhu yang nyaman (musim gugur).

---

### **Kesimpulan Umum**
Berdasarkan analisis data, dapat disimpulkan bahwa:
1. **Faktor Waktu:** Penyewaan sepeda paling tinggi pada sore hari (sekitar pukul 17:00) dan pada hari kerja (khususnya Jumat).
2. **Faktor Musim dan Cuaca:** Musim gugur dan suhu hangat (20-30°C) menjadi waktu terbaik untuk penyewaan sepeda.
3. **Faktor Pengguna:** Pengguna registered (berlangganan) mendominasi penyewaan sepeda, menunjukkan bahwa program membership sangat efektif.
4. **Faktor Hari Libur:** Hari libur tidak menjadi waktu yang populer untuk bersepeda, sehingga diperlukan strategi khusus untuk meningkatkan penyewaan di hari-hari tersebut.

### **Rekomendasi**
1. **Tingkatkan Promosi pada Waktu dan Musim Tertentu:**
   - Fokus pada sore hari (16:00-18:00) dan musim gugur untuk memaksimalkan penyewaan.
2. **Perluas Program Membership:**
   - Tawarkan manfaat tambahan seperti diskon atau akses eksklusif untuk menarik lebih banyak pengguna casual.
3. **Buat Program Khusus untuk Hari Libur:**
   - Sediakan paket keluarga atau tur bersepeda untuk menarik minat pengguna di hari libur.
4. **Tingkatkan Kenyamanan Pengguna:**
   - Sediakan fasilitas seperti tempat istirahat, minuman, atau penghangat selama musim dingin.
